server/app/lib/utils/cloud_forms/provider.rb

```diff
@@ -64,6 +64,32 @@ def self.add(cfme_ip, provider_params)
         }
         request_data = new_provider_form.request_data
         provider_added = agent.post("https://#{cfme_ip}/ems_infra/create/new?button=add", request_data, submit_headers)
+
+        host_list = agent.get("https://#{cfme_ip}/host/show_list", [], agent.page.uri)
```

Comment:
Does manageiq have REST APIs to support the operations being performed?  (e.g. host list, get, edit)

If not, can we create a bug/issue to have that functionality added?
If REST APIs are available, we may want to consider using them.  (This is on the premise that the ultimate long-term goal is to use APIs vs mechanize)